src/main/java/com/connexta/store/controllers/StoreController.java

```diff
@@ -129,6 +109,111 @@
     return ok().build();
   }
 
+  @Override
+  public ResponseEntity<Void> ingest(
+      String acceptVersion,
+      @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) OffsetDateTime lastModified,
+      MultipartFile file,
+      String correlationId,
+      MultipartFile metacard) {
+    if (lastModified == null || lastModified.toString().isBlank()) {
+      throw new ServerWebInputException(
+          String.format("%s is missing or blank", LAST_MODIFIED)); // TODO Replace this exception
+    }
+    String fileName = file.getOriginalFilename();
+    log.info("Ingest request received fileName={}", fileName);
+    InputStream inputStream;
+    InputStream metacardInputStream;
+    try {
+      inputStream = file.getInputStream();
+      metacardInputStream = metacard.getInputStream();
+    } catch (IOException e) {
+      throw new ValidationException("Could not open attachment"); // TODO Replace this exception
+    }
+    storeService.ingest(
+        file.getSize(),
+        file.getContentType(),
+        inputStream,
+        fileName,
+        metacard.getSize(),
+        metacardInputStream);
+
+    return ResponseEntity.accepted().build();
+  }
+
+  @ApiOperation(
```

Comment:
Not as part of this PR, but these should all be defined in the API repo, right?

Reply:
I don't think they need to be defined in the API. They would never be versioned and no client should ever construct the URI themselves.

Reply:
Why would a client need to construct the URL? Wouldn't they just say give me the metacard (should be more general in the future) for a given dataset?

Reply:
Our API doesn't work that way currently. No client ever constructs URLs, except clients of the ingest endpoint. Additionally, the retrieve endpoints are URIs, not URLs. They should never be versioned or have headers/params.